core/src/solution_submission.rs

```diff
@@ -246,18 +266,67 @@ fn extract_transaction_receipt(err: &MethodError) -> Option<&TransactionReceipt>
     }
 }
 
+fn convert_cancel_result(result: Result<(), NoopTransactionError>) -> SolutionSubmissionError {
+    match result {
+        Ok(()) => SolutionSubmissionError::Unexpected(anyhow!(
+            "solution submission transaction not confirmed in time"
+        )),
+        Err(err) => SolutionSubmissionError::Unexpected(
+            Error::from(err).context("failed to cancel solution submission"),
+        ),
+    }
+}
+
+trait IsNonceError {
+    fn is_nonce_error(&self) -> bool;
+}
+
+impl IsNonceError for ExecutionError {
+    fn is_nonce_error(&self) -> bool {
+        // This is the error as we've seen it on openethereum nodes.
+        // TODO: check how this looks on geth and infura. Not recognizing the error is not a serious
+        // problem but it will make us sometimes log an error when there actually was no problem.
+        match self {
+            ExecutionError::Web3(Web3Error::Rpc(RpcError { code, message, .. }))
+                if code.code() == -32010
+                    && message.find("Transaction nonce is too low.").is_some() =>
```

Comment:
I think this might be node dependant (especially the message).

Reply:
I agree. There is a comment about this right above.

Reply:
🙈

Reply:
Created https://github.com/gnosis/dex-services/issues/1196